torchvision/csrc/cpu/video/Video.cpp

```diff
@@ -0,0 +1,393 @@
+  return streamFPS.at(stream_str);
+}
+
+std::vector<double> Video::getDuration(std::string stream) const {
+  // add safety check
+  if (stream.empty()) {
+    stream = get<0>(current_stream);
+  }
+  auto stream_tpl = _parseStream(stream);
+  std::string stream_str = get<0>(stream_tpl);
+  // check if the stream exists
+  return streamDuration.at(stream_str);
+}
+
+void Video::Seek(double ts, bool any_frame = false) {
+  // initialize the class variables used for seeking and retrurn
+  video_any_frame = any_frame;
+  seekTS = ts;
+  doSeek = true;
+}
+
+torch::List<torch::Tensor> Video::Next(std::string stream) {
+  
+  bool newInit = false; // avoid unnecessary decoder initializations
+  if ((!stream.empty()) && (_parseStream(stream) != current_stream)) {
+      current_stream = _parseStream(stream);
+      newInit = true;
+  }
+
+  if ((seekTS != -1) && (doSeek == true)) {
+      newInit = true;
+      doSeek = false;
+  }
+
+  if (newInit){
+    succeeded = Video::_setCurrentStream();
+    if (succeeded) {
+      newInit = false;
+    }
+  }
+
+  // if failing to decode simply return a null tensor (note, should we
+  // raise an exeption?)
+  torch::Tensor framePTS = torch::zeros({1}, torch::kFloat);
+  torch::Tensor outFrame = torch::zeros({0}, torch::kByte);
+
+  // decode single frame
+  DecoderOutputMessage out;
+  int64_t res = decoder.decode(&out, decoderTimeoutMs);
+  // if successfull
+  if (res == 0) {
+    auto header = out.header;
+    const auto& format = header.format;
+
+    // initialize the output variables based on type
+    size_t expectedWrittenBytes = 0;
+
+    if (format.type == TYPE_VIDEO) {
+      // note: this can potentially be optimized
+      // by having the global tensor that we fill at decode time
+      // (would avoid allocations)
+      int outHeight = format.format.video.height;
+      int outWidth = format.format.video.width;
+      int numChannels = 3;
+      outFrame = torch::zeros({outHeight, outWidth, numChannels}, torch::kByte);
+      expectedWrittenBytes = outHeight * outWidth * numChannels;
+    } else if (format.type == TYPE_AUDIO) {
+      int outAudioChannels = format.format.audio.channels;
+      int bytesPerSample = av_get_bytes_per_sample(
+          static_cast<AVSampleFormat>(format.format.audio.format));
+      int frameSizeTotal = out.payload->length();
+
+      CHECK_EQ(frameSizeTotal % (outAudioChannels * bytesPerSample), 0);
+      int numAudioSamples =
+          frameSizeTotal / (outAudioChannels * bytesPerSample);
+
+      outFrame =
+          torch::zeros({numAudioSamples, outAudioChannels}, torch::kFloat);
+
+      expectedWrittenBytes = numAudioSamples * outAudioChannels * sizeof(float);
+    }
+    // currently not supporting other formats (will do soon)
+
+    // note: this will need to be revised to support less-accurate seek. So far keep as is
+    if (format.type == TYPE_VIDEO) {
+      auto numberWrittenBytes = fillVideoTensor(out, outFrame, framePTS);
```

Comment:
I know it's different from the current conventions, but I think I would prefer if we start returning Tensors in `CHW` format.
Can you add a `permute(2, 0, 1)` here?